Automattic/wp-calypso
# A4A: update unpaid invoice banner

Resolves https://github.com/Automattic/automattic-for-agencies-dev/issues/1230

## Proposed Changes

Currently, we have Payment notice banner available on Matketplace pages. However, it's generic at the moment. 
We want to make it more clear for users that they need to pay their bill. We are forcing this banner on all pages and make it non dismissible. We also have different messaging dependant on how many days overdue.

Here are banners:

<img width="1343" alt="Screenshot 2024-10-18 at 11 27 18 AM" src="https://github.com/user-attachments/assets/ac6d50cf-b5ad-47a1-a2c2-86b53f164d48">

<img width="1333" alt="Screenshot 2024-10-18 at 11 27 37 AM" src="https://github.com/user-attachments/assets/02107bf0-fec6-4c34-b365-f1049dfcb500">

<img width="1328" alt="Screenshot 2024-10-18 at 11 27 55 AM" src="https://github.com/user-attachments/assets/872e4f53-d605-450b-8842-09b9ac5caa45">

<img width="1328" alt="Screenshot 2024-10-18 at 11 28 12 AM" src="https://github.com/user-attachments/assets/f7da7fd0-50f8-4e27-9d9e-08d864f4b950">

<img width="1328" alt="Screenshot 2024-10-18 at 11 28 27 AM" src="https://github.com/user-attachments/assets/075df8a9-3e35-4652-851b-16fa27d6e6d6">


## Why are these changes being made?



## Testing Instructions



- You would need to patch this diff (D164082-code) to your sandbox (or use trunk if already merged). Point `public-api.wordpress.com` you your sandbox IP in your hosts file.
- The easiest way to test if you don't have overdue invoices:
 - In your sandbox, navigate to `wp-content/lib/jetpack-licensing/rest-api/class-invoice-resource.php` and set `status => 'open'`.
 - In the same file change `created` to some past date eg (`strtotime(Today - 3 days)`, `strtotime(Today - 20 days)`. 
 - Check different pages on live link of this pr, including `overview`, `marketplace` etc.

## Pre-merge Checklist
